### ISSUE TITLE
latex-mode does not always have correct functionality

### ISSUE THREAD

**Issue by samhedin** [syl20bnr/spacemacs#13647]
#### Description :octocat:
Sometimes when I open a .tex file, the proper latex mode does not appear to be activated.
For example, yesterday when I was editing a document the latex mode worked just as it should.
Today, opening the very same document results in some very reduced major mode being activated. (see: https://i.imgur.com/kEQ6EiX.png ). No updates were made to any part of the system between the time when it worked and now when it doesn't.
Someone on the spacemacs gitter suggested that this might be because of some naming business, since there is a major mode called latex-mode and one called LaTeX-mode, but switching between them does not appear to change anything. 

Removing my old .spacemacs, starting emacs to generate a new one, and then only adding the latex layer does not stop the issue from occurring. 

#### Reproduction guide :beetle:
I have not figured out how to reproduce this yet, but I am trying my best to figure it out.
It happens most of the time when I open a .tex-file. Even a new tex file in a new folder seems to cause the bug.

*Observed behaviour:* :eyes: :broken_heart:
A latex major mode with very reduced functionality.

*Expected behaviour:* :heart: :smile:
The latex major mode described in the documentation, which appears properly *sometimes*.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 28.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 23976ab79)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(go haskell yaml
    (clojure :variables clojure-enable-clj-refactor t clojure-enable-fancify-symbols t clojure-enable-linters 'clj-kondo)
    bibtex github
    (c-c++ :variables c-c++-backend 'lsp-ccls)
    (python :variables python-backend 'lsp python-format-on-save t python-pipenv-activate t)
    latex dap
    (lsp :variables lsp-rust-server 'rust-analyzer lsp-signature-render-documentation nil lsp-ui-doc-enable nil)
    (java :variables java-backend 'lsp)
    helm
    (auto-completion :variables haskell-completion-backend 'intero auto-completion-return-key-behavior nil auto-completion-tab-key-behavior 'nil auto-completion-enable-sort-by-usage t auto-completion-enable-help-tooltip 'manual)
    themes-megapack theming debug emacs-lisp git
    (multiple-cursors :variables multiple-cursors-backend 'evil-mc)
    treemacs org
    (rust :variables rust-backend 'lsp)
    (shell :variables shell-default-shell 'multi-term shell-default-height 30 shell-default-position 'bottom)
    (syntax-checking :variables syntax-checking-enable-tooltips nil))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE HARFBUZZ LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP

**Comment by smile13241324** [comment]
@samhedin I have seen you are using emacs 28, can you try to reproduce on emacs 26.3 and post the results please? Would also be interesting if there are any error messages in your message buffer when you have the reduced latex mode.

Also just to rule this out, please pull the latest version from develop and update your emacs packages.

From the screenshot I would assume that the loading process did throw an error leaving you with a partial config however I would not know why this would only happen von time to time.

**Comment by samhedin** [comment]
I should clearly have thought of this before... I even running https://aur.archlinux.org/packages/emacs-native-comp-git/
The issue does not seem to occur with any other versions. I tried 26, 27, and 28 without native comp and they all appear to work. Something I noticed, in the other version the major mode is shown as `LaTeX/MPS` instead of just `LaTex`. What's that about?

I will keep an eye out and see if other people encounter the same issue. Maybe this is an issue with the native comp branch, or I am just doing something wrong. I'll try to figure out which.
